_posts/2019-10-27-Merging-Multiple-SimpleCov-coverage-results.markdown

```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+title: Merging-Multiple-SimpleCov-coverage-results
+date: 2019-10-29 13:56:00
+categories: ["Rails", "Simplecov", "Upgrades"]
+author: bronzdoc
+---
+
+As part of our Roadmap service at FastRuby.io, we have to analyze the test suite of the application we are upgrading to give a proper estimate on how long it will take us to upgrade. We use SimpleCov for this.
+We got in to the problem of having SimpleCov results across different containers for the same application. Our goal was to have a single result for the whole application, so in this blog post we are going to show you how we solved that problem.
+
+<!--more-->
+
+Some of the applications we upgrade are outdated and setups can be difficult, with no documentation, missing steps on how to set it up, etc. Even after the setup, the tests of these applications can take several hours to complete and generate a coverage report.
+
+So, we have been approaching the problem a little different. Instead of executing the tests and generating the report locally, we rely on their CI configuration to get the coverage data and move on with the estimates. After all, the coverage report is just a metric to give us an idea on how much effort it will take us to complete the upgrade.
```

Comment:
This sounds a little weird:

```
So, we have been approaching the problem a little different.
```

Maybe you mean:

```
So, we recently decided to take a different approach for this problem.
```

And:

```
Instead of executing the tests and generating the report locally, we rely on their CI configuration to get the coverage data and move on with the estimates.
```

Sounds better:

```
Instead of executing the tests locally and generating the report, we rely on our client's CI configuration to get the coverage data and move on.
```